code/modules/mob/living/carbon/taste.dm

```diff
@@ -0,0 +1,60 @@
+
+/mob/living/carbon/proc/ingest(var/datum/reagents/from, var/datum/reagents/target, var/amount = 1, var/multiplier = 1, var/copy = 0) //we kind of 'sneak' a proc in here for ingesting stuff so we can play with it.
+	if(last_taste_time + 50 < world.time)
+		var/datum/reagents/temp = new(amount) //temporary holder used to analyse what gets transfered.
+		from.trans_to_holder(temp, amount, multiplier, 1)
+
+		var/text_output = temp.generate_taste_message(src)
+		if(text_output != last_taste_text || last_taste_time + 200 < world.time) //We dont want to spam the same message over and over again at the person. Give it a bit of a buffer.
+			src << "<span class='notice'>You can taste [text_output]</span>" //no taste means there are too many tastes and not enough flavor.
+			last_taste_time = world.time
+			last_taste_text = text_output
+	return from.trans_to_holder(target,amount,multiplier,copy) //complete transfer
+
+/* what this does:
+catalogue the 'taste strength' of each one
+calculate text size per text.
+*/
+/datum/reagents/proc/generate_taste_message(mob/living/carbon/taster = null)
+	var/minimum_percent = 15
+	if(ishuman(taster))
```

Comment:
reagents are present up to mob/living, monkeys apparently cant taste anything nor can xenos despite being able to ingest reagents